### PR TITLE
feat: add new transaction handler

### PR DESCRIPTION
**Pull request by abc3**

### CODE REVIEW

File: lib/supavisor/handlers/proxy/handler.ex [hunk 1]
```diff
@@ -146,9 +145,13 @@ defmodule Supavisor.Handlers.Proxy.Handler do
     HH.sock_close(data.sock)
     HH.sock_close(data.db_sock)
 
-    case Registry.lookup(Supavisor.Registry.TenantClients, data.id) do
-      clients when clients == [{self(), []}] or clients == [] -> PromEx.remove_metrics(data.id)
-      _ -> :ok
+    if data.id != nil do
+      with clients <- Registry.lookup(Supavisor.Registry.TenantClients, data.id),
```
Comment (hauleth) [inline lib/supavisor/handlers/proxy/handler.ex:149]: Why use `with` for this clause? It cannot fail.

File: lib/supavisor.ex [hunk 2]
```diff
@@ -349,4 +349,32 @@ defmodule Supavisor do
...
+  end
+
+  @spec start_local_server(map()) :: {:ok, map()} | {:error, any()}
+  def start_local_server(args) do
+    acceptors = round(args.max_clients / 100)
+
+    opts =
+      %{
+        max_connections: args.max_clients,
+        num_acceptors: if(acceptors < 10, do: 10, else: acceptors),
```
[the remark this answers is not in the document]
Comment (abc3) [reply lib/supavisor.ex:368]: ohh, cool, didn't know  that

File: lib/supavisor/handlers/proxy/client.ex [hunk 3]
```diff
@@ -4,20 +4,17 @@ defmodule Supavisor.Handlers.Proxy.Client do
   require Logger
 
   alias Supavisor, as: S
-  alias Supavisor.ProxyDb, as: Db
   alias Supavisor.Helpers, as: H
   alias Supavisor.HandlerHelpers, as: HH
 
-  alias Supavisor.{
+  alias S.{
```
[the remark this answers is not in the document]
Comment (abc3) [reply lib/supavisor/handlers/proxy/client.ex:10]: yeah, let's make it more readable

File: lib/supavisor/handlers/proxy/handler.ex (hunk 1 above)
Comment (abc3) [reply lib/supavisor/handlers/proxy/handler.ex:149]: :facepalm:

File: lib/supavisor.ex [hunk 4]
```diff
@@ -349,4 +349,36 @@ defmodule Supavisor do
...
+      _ -> {:error, :not_found}
+    end
+  end
+
+  @spec start_local_server(map()) :: {:ok, map()} | {:error, any()}
+  def start_local_server(%{max_clients: max_clients} = args) do
+    # max_clients=-1 is used for testing the maximum allowed clients in ProxyTest
+    {acceptors, max_clients} =
+      if max_clients > 0,
+        do: {round(max_clients / 100), max_clients},
```
Comment (hauleth) [inline lib/supavisor.ex:366]: What if `max_clients` is smaller than `50`? We will end with no `acceptors`?

**Review by hauleth — APPROVED** [review]

File: lib/supavisor.ex (hunk 4 above)
Comment (abc3) [reply lib/supavisor.ex:366]: right, must be `ceil`

File: lib/supavisor/db_handler.ex [hunk 5]
```diff
@@ -31,6 +31,12 @@ defmodule Supavisor.DbHandler do
   @spec cast(pid(), pid(), binary()) :: :ok | {:error, any()} | {:buffering, non_neg_integer()}
   def cast(pid, caller, msg), do: :gen_statem.cast(pid, {:db_cast, caller, msg})
 
+  @spec set_idle(pid()) :: :ok
+  def set_idle(pid), do: :gen_statem.cast(pid, :set_idle)
+
+  @spec change_skt_owner(pid(), pid()) :: {:ok, S.sock()}
+  def change_skt_owner(pid, caller), do: :gen_statem.call(pid, {:tcp_owner, caller}, 15_000)
```
Comment (hauleth) [inline lib/supavisor/db_handler.ex:38]:
```suggestion
  @spec change_socket_owner(pid(), pid()) :: {:ok, S.sock()}
  def change_socket_owner(pid, caller), do: :gen_statem.call(pid, {:tcp_owner, caller}, 15_000)
```

Do not shorten name as it make it less readable.

Comment (hauleth) [reply lib/supavisor/db_handler.ex]: If you want to shorten it, then `sock` would be IMHO better than `skt`, and it is only 1 character longer.

**Pull request merged**